doc/tutorials/chapter_8/01_interface.md

```diff
@@ -0,0 +1,176 @@
+      {this.width = 8, super.name = 'counter'}) {
+    // Register inputs and outputs of the module in the constructor.
+    // Module logic must consume registered inputs and output to registered
+    // outputs.
+    en = addInput('en', en);
+    reset = addInput('reset', reset);
+    clk = addInput('clk', clk);
+
+    final val = addOutput('val', width: width);
+
+    // A local signal named 'nextVal'.
+    final nextVal = Logic(name: 'nextVal', width: width);
+
+    // Assignment statement of nextVal to be val+1
+    // ('<=' is the assignment operator).
+    nextVal <= val + 1;
+
+    // `Sequential` is like SystemVerilog's always_ff, in this case trigger on
+    // the positive edge of clk.
+    Sequential(clk, [
+      // `If` is a conditional if statement, like `if` in SystemVerilog
+      // always blocks.
+      If(reset, then: [
+        // The '<' operator is a conditional assignment.
+        val < 0
+      ], orElse: [
+        If(en, then: [val < nextVal])
+      ])
+    ]);
+  }
+}
+```
+
+## Counter Module Interface
+
+Let us see how we can change the `ROHD` module to `Counter` interface. First, we can create a enum `CounterDirection` that have tags of `inward`, `outwards` and `misc`. You can think of this as what is the category you want to group your ports. This category can be reuse between modules. `inwards` port group all inputs port, `outwards` group all outputs port and `misc` group all miscellanous port such as `clk`.
+
+Then, we can create our interface `CounterInterface` that extends from parents `Interface<TagType>`. The `TagType` is the enum that we create earlier. Let create the getters to all ports for `Counter` to allows us to send signals to the interface.
+
+As normal, we will need to create the constructor `CounterInterface`. Inside the constuctor, add `setPorts()` function to group our common port. `setPorts` takes have function sinature of `void setPorts(List<Logic> ports, [List<CounterDirection>? tags])` which received a List of `Logic` and tags.
+
+Hence, the `CounterInterface` will look something like this:
+
+```dart
+enum CounterDirection { inward, outward, misc }
+
+/// A simple [Interface] for [Counter].
+class CounterInterface extends Interface<CounterDirection> {
+  Logic get en => port('en');
+  Logic get reset => port('reset');
+  Logic get val => port('val');
+  Logic get clk => port('clk');
+
+  final int width;
+  CounterInterface({this.width = 8}) {
+    setPorts([Port('en'), Port('reset')], [CounterDirection.inward]);
+
+    setPorts([
+      Port('val', width),
+    ], [
+      CounterDirection.outward
+    ]);
+
+    setPorts([Port('clk')], [CounterDirection.misc]);
+  }
+}
+```
+
+Next, we want to modify the `Counter` module constructor to receive the interface. Then, we **MUST** create a new instance of the interface to avoid modify the interface inside the constructor.
+
+```dart
+// create a new interface instance.
+late final CounterInterface intf;
```

Comment:
might be better to mark this `_intf`, then you can avoid the `this.` down below.  If you make fixes to the example, feel free to submit them to rohd/rohd-vf :)